### PR TITLE
[KSP] Exact intersections bugfix

### LINKED ISSUE #8624: [KSP] Finalizing Partition Error [Help][CGAL 6.0.1]
When I use KSP(Kinetic Shape Partition)[CGAL 6.0.1] to partition the space, the code `ksp.partition(2)` reports an error.

## Issue Details

Details of the error are as follows:

```
--- FINALIZING PARTITION:
* getting volumes ...
CGAL error: precondition violation!
Expression : !collinear(o_prime,o,q)
File       : D:\Program_Files\CGAL-6.0.1\CGAL-6.0.1\include\CGAL/Polygon_mesh_processing/internal/Corefinement/predicates.h
Line       : 77
Explanation:
Refer to the bug-reporting instructions at https://www.cgal.org/bug_report.html
ERROR: constrained Delaunay triangulation failed
```

I found that in `sorted_around_edge` function, points `o_prime, o, q` were detected as collinear errors. I would like to ask how can I add constraints to the `input_vertices` and `input_faces` to avoid such errors?

I hope someone can give me some ideas. Thank you very much~

## Source Code

```
using EPICK = CGAL::Exact_predicates_inexact_constructions_kernel;
using EPECK = CGAL::Exact_predicates_exact_constructions_kernel;

using Kernel = EPICK;
using FT = typename Kernel::FT;
using Point_3 = typename Kernel::Point_3;

using Surface_mesh = CGAL::Surface_mesh<Point_3>;
using KSP = CGAL::Kinetic_space_partition_3<EPICK>;
using Timer = CGAL::Real_timer;

std::vector<Point_3> input_vertices;
std::vector<std::vector<std::size_t> > input_faces;

try
	{
		KSP ksp(CGAL::parameters::verbose(true).debug(false));
		ksp.insert(input_vertices, input_faces);

		ksp.initialize(CGAL::parameters::bbox_dilation_ratio(1.1).reorient_bbox(false));

		ksp.partition(2);  // Error reported here!!

		ksp.getVolumes<EPICK>(output_vertices, output_faces, output_volume_centers);
	}
	catch (std::exception& e) {
		std::cerr << "\t--STOCKER: partition failed: " << e.what() << std::endl;
		return false;
	}
	catch (...) {
		return false;
	}

	if (output_faces.size() != output_volume_centers.size()) {
		return false;
	}
```

## Environment

* Operating system (Windows, 64 bits):
* Compiler: visiual studio 2022
* Release or debug mode: Debug
* CGAL version: 6.0.1
* Boost version: 1.79.0
* Other libraries versions if used: Eigen-3.3.4

### PR DESCRIPTION
**Pull request by soesau**
## Summary of Changes

Switching initial_polygon_iedge_intersections() and calculate_edge_intersection_time() to Intersection_kernel (epeck)
Resolving potential non-convex partitions

## Release Management

* Affected package(s): KSP
* Issue(s) solved (if any): fix #8624

### CODE REVIEW

File: Kinetic_space_partition/include/CGAL/KSP_3/Initializer.h [hunk 1]
```diff
@@ -499,7 +493,7 @@ class Initializer {
 
                 typename Intersection_graph::Kinetic_interval& kinetic_interval = m_data.igraph().kinetic_interval(e, sp_idx);
                 crossing_iedges.push_back(e);
-                if (emin > s) {
+                if (emin > s || (min_speed != min_speed)) {
```
Comment (albert-github) [inline Kinetic_space_partition/include/CGAL/KSP_3/Initializer.h:496]: Is this for testing against e.g. NaN or ...?
In the first case it is probably clearer to use `!std::isNaN(min_speed)`

File: Kinetic_space_partition/include/CGAL/KSP_3/Initializer.h [hunk 2]
```diff
@@ -511,7 +505,7 @@ class Initializer {
                   kinetic_interval.push_back(std::pair<FT, FT>(0, 0));
                 }
 
-                if (t > emax) {
+                if (t > emax || (max_speed != max_speed)) {
```
Comment (albert-github) [inline Kinetic_space_partition/include/CGAL/KSP_3/Initializer.h:508]: see `min_speed`

File: Kinetic_space_partition/include/CGAL/KSP_3/Initializer.h (hunk 1 above)
Comment (albert-github) [reply Kinetic_space_partition/include/CGAL/KSP_3/Initializer.h]: By head I gave `isnan` as possibility, but maybe it is better to use `isfinite`, also checkout `isinf` and `isnormal`

Comment (soesau) [reply Kinetic_space_partition/include/CGAL/KSP_3/Initializer.h]: Thank you. isinf is actually the correct one.

**Comment by sloriot** [comment]
Successfully tested in CGAL-6.1-Ic-33

**Comment by sloriot** [comment]
@soesau Please check User manual timings are still relevant.

**Pull request merged**